src/main.cpp

```diff
@@ -0,0 +1,234 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of validator-keys-tool:
+        https://github.com/ripple/validator-keys-tool
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <beast/unit_test/dstream.hpp>
+#include <boost/program_options.hpp>
+#include <ripple/beast/unit_test.h>
+#include <ValidatorKeys.h>
+
+namespace po = boost::program_options;
+
+void printHelp (const po::options_description& desc)
+{
+
+    static std::string const name = "ripple_validator_keys";
+
+    std::cerr
+        << name << " [options] <command> <params>\n"
+        << desc << std::endl
+        << "Commands: \n"
+           "     create_master_keys\n"
+           "     create_signing_keys\n"
+           "     revoke_master_keys\n";
+}
+
+static int runUnitTests ()
+{
+    using namespace beast::unit_test;
+    beast::unit_test::dstream dout{std::cout};
+    reporter r{dout};
+    bool const anyFailed = r.run_each(global_suites());
+    if(anyFailed)
+        return EXIT_FAILURE;
+    return EXIT_SUCCESS;
+}
+
+void createKeyFile (const boost::filesystem::path& keyFile)
+{
+    using namespace boost::filesystem;
+    using namespace ripple;
+
+    if (exists (keyFile))
+    {
+        throw std::runtime_error (
+            "Refusing to overwrite existing key file: " +
+                keyFile.string ());
+    }
+
+    ValidatorKeys keys (KeyType::ed25519);
+    keys.writeToFile (keyFile);
+
+    std::cout << "Master validators stored in " <<
+        keyFile.string() << std::endl;
+}
+
+void signManifest (const boost::filesystem::path& keyFile,
+    boost::optional<std::uint32_t> sequence)
+{
+    using namespace ripple;
+
+    auto keys = ValidatorKeys::make_ValidatorKeys (keyFile);
+
+    if (sequence)
+    {
+        if (*sequence <= keys.sequence)
+            throw std::runtime_error (boost::str (boost::format (
+                "Sequence should exceed current sequence (%u).") %
```

Comment:
`boost::str` and `boost::format` need `#include <boost/format.hpp>`. Not sure how that's working on the non-Windows builds.